apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,350 @@
+
+For example,
+
+- `Slot<'div'>` means by default the slot will render a `div` element.
+- `Slot<typeof Button>` means by default the slot will render a `Button` component.
+- `Slot<'span', 'div' | 'pre'>` means that caller can use `as`to change from `span` to `div` or `pre`.
+
+### Slot definition
+
+A `Slot` can be passed a literal value, an object containing props, or a render function callback.
+The `Slot` type is defined using `WithSlotShorthandValue<Props>`.
+
+> The type definition here is simplified.
+> For full details, see the `compose/types.ts` file comments.
+
+```ts
+type Slot<Type, AlternateAs> =
+      ...
+      WithSlotShorthandValue<
+          Type extends keyof JSX.IntrinsicElements
+            ? { as?: Type } & WithSlotRenderFunction<IntrisicElementProps<Type>>
+            : Type extends React.ComponentType<infer Props>
+            ? WithSlotRenderFunction<Props>
+            : Type
+        >
+      ...
+```
+
+Following `WithSlotShorthandValue<Props>`, it is defined as the `Props` object
+or the `children` of a React child, array of nodes, or portal.
+
+```ts
+type WithSlotShorthandValue<Props extends { children?: unknown }> =
+  | Props
+  | Extract<SlotShorthandValue, Props['children']>;
+
+type SlotShorthandValue = React.ReactChild | React.ReactNodeArray | React.ReactPortal;
+```
+
+Following `WithSlotRenderFunction<Props>`, it is defined as the `Props` object and children.
+The children can be JSX/TSX or a function that renders JSX/TSX.
+
+```ts
+type WithSlotRenderFunction<Props extends { children?: unknown }> = Props & {
+  children?: Props['children'] | SlotRenderFunction<Props>;
+};
+
+type SlotRenderFunction<Props> = (
+  Component: React.ElementType<Props>,
+  props: Omit<Props, 'children' | 'as'>,
+) => React.ReactNode;
+```
+
+### Defining slots in component props
+
+In the Fluent UI React composition architecture, components define their props using `ComponentProps<Slot>`
+and state for rendering using `ComponentState<Slot>`.
+
+For example, Spinner defines optional spinner and label slots.
+
+```tsx
+type SpinnerSlots = {
+  root: NonNullable<Slot<'div'>>;
+  spinner?: Slot<'span'>;
+  label?: Slot<typeof Label>;
+};
+```
+
+#### Optional slots
+
+Like any property, slots can be made optional by using the trailing `?`.
+The component developer can decide if default content is rendered when a slot is `undefined` or
+if nothing is rendered.
+
+#### NonNullable slots
+
+When a slot is defined in the props of a component, it can be set to null to prevent the slot from being rendered.
+In cases where components need to require a slot to always be rendered,
+it can be wrapped in `NonNullable<T>`.
+
+For example, in `RadioButton` the indicator must always be rendered.
+
+```ts
+indicator: NonNullable<Slot<'div'>>;
+```
+
+Non-nullable and optional/required slots are independent from one another.
+A slot can be defined with as optional or required and nullable or non-nullable.
```

Comment:
These 2 are great sections! I've often found that partners struggle a lot with the difference between these two. Thanks for making it so clear!